presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -332,7 +345,51 @@ private PlanBuilder appendQuantifiedComparisonApplyNode(PlanBuilder subPlan, Qua
             case GREATER_THAN_OR_EQUAL:
                 return planQuantifiedOrderable(subPlan, quantifiedComparison, correlationAllowed);
         }
-        throw new SemanticException(NOT_SUPPORTED, quantifiedComparison, "Quantified comparison with the given quantifier and comparison type is not supported");
+        // all cases are checked, so this exception should never be thrown
+        throw new IllegalStateException("Unexpected combination of quantifier and comparison type: " +
+                quantifiedComparison.getQuantifier() + ", " + quantifiedComparison.getComparisonType());
+    }
+
+    private PlanBuilder planQuantifiedEqualsAll(PlanBuilder subPlan, QuantifiedComparisonExpression quantifiedComparison, boolean correlationAllowed)
+    {
+        checkArgument(quantifiedComparison.getComparisonType() == EQUAL && quantifiedComparison.getQuantifier() == Quantifier.ALL);
+        // A = ALL B <=> min B = max B && A = min B
+        RelationPlan subqueryRelationPlan = createRelationPlan(quantifiedComparison.getSubquery());
+        PlanNode subqueryPlan = subqueryRelationPlan.getRoot();
+        List<Expression> outputColumnReference = ImmutableList.of(getOnlyElement(subqueryRelationPlan.getOutputSymbols()).toSymbolReference());
+        Type outputColumnType = getOnlyElement(subqueryRelationPlan.getDescriptor().getAllFields()).getType();
+        if (!outputColumnType.isOrderable()) {
+            throwNotSupportedException(quantifiedComparison, "Quantified comparison with given parameters on unorderable type");
```

Comment:
I think it should be possible to implement this without requiring the type to be orderable. 

`A = ALL B <=> A = arbitrary B && count(distinct B) = 1`

We might be able to rewrite the query along these lines:

```
x = (
    SELECT arbitrary(y)
    FROM B
    HAVING count(DISTINCT y) = 1
)
```

Computing `count(DISTINCT y)` could get expensive. We could further improve it by doing something like this, which takes advantage of the distinct-limit optimization.

```
x = (
    SELECT arbitrary(y) 
    FROM (
        SELECT DISTINCT y
        FROM B
        LIMIT 2
    )
    HAVING count(DISTINCT y) = 1
)
```